newdust/scatteringmodel/make_ggadt.py

```diff
@@ -0,0 +1,243 @@
+
+    outfile: string: name of the ouputted FITS file
+
+    overwrite: bool: whether or not to overrite a preexisting FITS file with the same name
+
+    Returns FITS file with the data described above
+    """
+
+    #data to store in FITS
+    radii = []
+    qext = []
+    qabs = []
+    qsca = []
+    diff = [] #NEED TO IMPLEMENT
+
+    #constant parameters
+    evs = []
+    theta = [0.0] #NEED TO IMPLEMENT
+    axis_ratio = 0.0
+    shape = ''
+    orientation = ''
+
+    #go through the output files and popuate above variables
+    for i in indicies:
+        filename= f'{folder}/{material}_{i}.out'
+        data = parse_file(filename)
+
+        qext.append(data['qext'])
+        qabs.append(data['qabs'])
+        qsca.append(data['qsca'])
+        diff.append(data['diff'])
+        radii.append(data['radius'])
+
+        #need to either assign or check consistency of constant parameters
+        if not evs: 
+            evs = data['evs']
+        elif evs != data['evs']:  
+            raise Exception('Error: Incident energy range must be constant')
+
+        if axis_ratio == 0.0:
+            axis_ratio = data['axis ratio']
+        elif axis_ratio != data['axis ratio']:
+            raise Exception('Error: Axis ratio must be constant')
+        
+        if shape == '':
+            shape = data['shape']
+        elif shape != data['shape']:
+            raise Exception('Error: Shape must be constant')
+        
+        if orientation == '':
+            orientation = data['orientation']
+        elif orientation != data['orientation']:
+            raise Exception('Error: Orientation must be constant')
+    
+    #Radii and indicies should be the same length -- if they're not something is wrong
+    if len(indicies) != len(radii):
+        raise Exception('Error: Too many radii') if len(indicies) < len(radii) else Exception('Error: Not enough radii')
+
+    #make parameters and header
+    header = make_header(material, shape, axis_ratio, orientation)
+    pars = make_pars(evs, radii, theta)
+    
+    #need to transpose the shape of qext, qabs, qsca, and diff to follow scatteringmodel
+    #diff also needs the extra dimension for theta
+    qext = np.array(qext).transpose()
+    qabs = np.array(qabs).transpose()
+    qsca = np.array(qsca).transpose()
+    diff = np.expand_dims(np.array(diff).transpose(), 2)
+    
+    img_list = []
+    for (val, head) in zip([qext, qabs, qsca, diff],
+                           ["Qext", "Qabs", "Qsca", "Diff-xsect (ster^-1)"]):
+        htemp = fits.Header()
+        htemp['TYPE'] = head
+        img_list.append(fits.ImageHDU(val, header=htemp))
+
+    #write table
+    fnl_list = [header] + pars + img_list
+    hdu_list = fits.HDUList(hdus=fnl_list)
+    hdu_list.writeto(outfile, overwrite=overwrite)
+    return outfile
+
+#Helper functions:
```

Comment:
Can you rename all helper functions across make_ggadt.py and make_ggadt_astrodust.py to start with an underscore (e.g., `_parse_file`)

Python doesn't implement private and public variables and functions in exactly the same way as other languages like C, and my understanding in general is that people add an underscore to the beginning in order to signal to the user that this isn't a function intended for general use. That doesn't mean we won't or can't access these functions at some point, but the underscore signals the intention.